tests/test_completion.py

```diff
@@ -236,172 +78,223 @@ def move_cursor(self, row: int, col: int) -> None:
         s.add(point)
 
     def select_completion(self) -> 'Generator':
+        current_change_count = self.view.change_count()
         self.view.run_command('auto_complete')
+        committed = False
 
-        yield 100
-        self.view.run_command("commit_completion")
+        def commit_completion() -> bool:
+            if not self.view.is_auto_complete_visible():
+                return False
+            nonlocal committed
+            if not committed:
+                self.view.run_command("commit_completion")
+                committed = True
+            return self.view.change_count() > current_change_count
+
+        yield commit_completion
 
     def read_file(self) -> str:
         return self.view.substr(sublime.Region(0, self.view.size()))
 
-    def test_simple_label(self) -> 'Generator':
-        self.set_response("textDocument/completion", completions_with_label)
-
-        self.type("a")
+    def verify(self, response: List[Dict[str, Any]], expected_text: str,
```

Comment:
What do you think about reordering the arguments?
instead of 
`response, expected_text, pre_insert_text`
to be: 
`response, insert_text, expected_text`?

And do you think that this:
```
self.verify([{'label': 'asdf', 'insertText': 'asdf()'}], 
            insert_text='a', 
            expected_text='asdf()')
```
is more easier to understand on first glance, than when you see this?
`self.verify([{'label': 'asdf', 'insertText': 'asdf()'}], 'asdf()', "a")`